### ISSUE TITLE
Support CSS4 length units

### ISSUE THREAD

**Issue by RazrFalcon** [RazrFalcon/resvg#412]
https://www.w3.org/TR/css-values/#relative-lengths

- [ ] `ch`
- [ ] `rem`
- [ ] `vw`/`vh`
- [ ] `vmin`/`vmax`
- [ ] `Q`

Looks like no one supports `cap`, `ic`, `lh`, `rlh`, `vi`, `vb` yet, so we ignore them too.

**Comment by LaurenzV** [comment]
If I remember correctly, when parsing those lengths we don't have access to the font metrics, right? For the `ch` and `ex` units, would it be acceptable for you to just use 0.5em and 1em (since this seems to explicitly be allowed by the specification)?

<img width="818" alt="image" src="https://github.com/RazrFalcon/resvg/assets/47084093/045f33b1-6954-41c5-b826-7775424f26fa">

**Comment by RazrFalcon** [comment]
Yes, this is kinda chicken and egg problem. The way usvg works is that we parse SVG first, then then user scans it for text nodes and loads required fonts to the fontdb and then converts text to paths using this db. 
Sure, we can just use system fonts during parsing, but it would be convoluted and fragile. Not to mention the performance hit form scanning system fonts. And also, there are no "system fonts" in WASM.

Therefore hardcoded values are our only choice. 

PS: SVG 2 also supports embedded fonts, but we don't, yet. Which is a whole another can of worms.
